### PR TITLE
Add gpt example

### PR DESCRIPTION
**Pull request by husimplicity**
1. Add an example which tests the capability of GPT on graph tasks
2. Fix negative sampler sharing lock with sampler
3. rename build.py to build_glt.py to avoid potential duplication of name

### CODE REVIEW

**Comment by LiSu** [comment]
Please add a README introducing the GPT example.

**Comment by husimplicity** [comment]
> Please add a README introducing the GPT example.

readme Added

File: examples/gpt/utils.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  return message
+
+
+def link_prediction(batch, titles, reason=False):
+  message = "This is a directed subgraph of arxiv citation network with " + str(batch.x.shape[0]) + " nodes numbered from 0 to " + str(batch.x.shape[0]-1) + ".\n"
+  graph = batch.edge_index.T.unique(dim=0).tolist()
+  message += "The titles of each paper:\n"
+  for i in range(batch.x.shape[0]):
+    message += "node " + str(i) + " is '" + titles[i][0] + "'\n" 
+  message += "The sampled subgraph of the networks are " + str(graph) + ' where the first number indicates source node cites the second destination node.\n'
```
Comment (LiSu) [inline examples/gpt/utils.py:52]: Clarify

File: examples/gpt/utils.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  message += "Hint: the direction of the edge can indicate some information of temporality.\n"
+  message += "\nAccording to principle of the construction of citation networks and the given subgraph structure, answer the following questions:\n"
+  message += "Question 1: predict whether there tends to form an edge "+str(batch.edge_label_index.T.tolist()[1])+".\n"
```
Comment (LiSu) [inline examples/gpt/utils.py:55]: add comment

File: examples/gpt/arxiv.py [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  titles = pd.read_csv(root + "titles.csv.gz").to_numpy()
+  ids = torch.from_numpy(pd.read_csv(root + "ids.csv.gz").to_numpy())
+  edge_index = torch.from_numpy(pd.read_csv(root + "edges.csv.gz").to_numpy())
+
+  print('Build graphlearn_torch dataset...')
+  start = time.time()
+  glt_dataset = glt.data.Dataset()
+  glt_dataset.init_graph(
+    edge_index=edge_index.T,
+    graph_mode='CPU',
```
Comment (LiSu) [inline examples/gpt/arxiv.py:66]: As the data is stored in CPU memory, it's reasonable to use CPU sampling instead of GPU sampling

Comment (husimplicity) [reply examples/gpt/arxiv.py:65]: done

File: examples/gpt/utils.py (hunk 2 above)
Comment (husimplicity) [reply examples/gpt/utils.py:58]: added

File: examples/gpt/arxiv.py [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    # print(message)
+    response = get_gpt_response(
+      message=message
+    )
+
+    print(f"response: {response}")
+
+
+if __name__ == '__main__':
+  world_size = torch.cuda.device_count()
```
[the remark this answers is not in the document]
Comment (husimplicity) [reply examples/gpt/arxiv.py]: deleted

File: examples/gpt/utils.py (hunk 1 above)
Comment (husimplicity) [reply examples/gpt/utils.py]: done

**Pull request merged**